src/components/Table.js

```diff
@@ -248,18 +257,28 @@ export default class Table extends React.Component<TableDefaultProps, TableProps
       const anchorRowIndex = sortedRows.length > 0 ? 0 : null;
       const activeRowIndex = sortedRows.length > 0 ? 0 : null;
 
+      // If the parent provided an update hook, update the parent with the changes.
+      if (newProps.onSelect) {
+        const selectedRows = activeRowIndex ? [sortedRows[activeRowIndex]] : [];
+        const selectedRow = activeRowIndex ? sortedRows[activeRowIndex] : null;
+        newProps.onSelect(selectedRows, selectedRow);
+      }
       this.setState({
         sortedRows,
         selectedRowIndices,
         anchorRowIndex,
         activeRowIndex,
       });
-
-      // If the parent provided an update hook, update the parent with the changes.
-      if (newProps.onSelect) {
-        const selectedRows = sortedRows.length > 0 ? [0] : [];
-        newProps.onSelect(selectedRows, activeRowIndex);
-      }
+    } else if (!isEqual(newProps.rows, this.props.rows)) {
+      // Selection hasn't changed, but other row data has update row data
+      const sortedRows = newProps.rows && newProps.rows.length > 0
+      ? newProps.rows.map((row, index) => {
+        return { ...row, index };
```

Comment:
Actually, another question: We should either document for the rows parameter that the objects cannot have a field named "index" or we should maybe rename the index field we're adding to the row objects to something more esoteric ('tableComponentIndex' or something?).

Antother caveat regarding the row objects is that the onSelect method is being passed the value from the sortedRows array and not the original object passed in. So if I pass an array of MyClass objects and I have a doSomething() method on MyClass, I won't be able to call row.doSomething() because the copy created with the spread operator won't have the prototype included, if I remember correctly.... I think we could probably just document this, too—as long as people are told explicitly, they can probably work around it.

Reply:
Because of the need for immutability in prop and state updates, more and more people are getting away from using classes for passing data around and instead using plain old js objects, especially with ES6. redux and functional programming practices have especially increased that mentality. That said, I agree I should give `index` a different name. Currently, the doc say a row must be an object, I'll update it to say it will be treated as an object.

Reply:
OK